actnow/accounts/signals.py

```diff
@@ -3,10 +3,13 @@
 from django.dispatch import receiver
 from rest_framework.authtoken.models import Token
 
+from actnow.profiles.models import UserProfile
+
 User = get_user_model()
 
 
 @receiver(post_save, sender=User)
 def create_auth_token(sender, instance=None, created=False, **kwargs):
     if created:
         Token.objects.create(user=instance)
+        UserProfile.objects.create(user=instance, **instance.extra_fields)
```

Comment:
Here is where we can set `first_name` to `instance.username` if `extra_fields.first_name` is None.